### DISCUSSION TITLE
Getting PoE speakers to play audio files via MQTT

### DISCUSSION THREAD

**Discussion by ifordesmith** [AxisCommunications/discussions#34] [VAPIX] [answered]
Hi, does anyone know if its possible to get a speaker to subscribe to an MQTT topic and send it an audio file (mp3 or stream) to play ?

Many thanks in advance

**Comment by mattias-kindborg-at-work** [comment]
Hi @ifordesmith.

No, I don't think you can do that without having to write an [ACAP](https://www.axis.com/products/acap). What you can do, without having to write a single line of code, is to use [events](https://help.axis.com/get-started-with-rules-for-events) on the device to configure a rule that is triggered by a specific MQTT topic, than then plays an audio clip that previously has been uploaded to the device.

The requirements for MQTT subscriptions is that the device must use firmware version 10.7.0 or higher.

**Comment by sproit** [comment]
I've been looking for the exact same functionality. But I noticed that there is a huge websocket API included with the Audio Manager Edge, which is unfortunately undocumented. Any chance we could get some docs? Being able to switch schedules, or simply launch any stream programmatically would be a big improvement.

**Reply by mattias-kindborg-at-work** [reply]
Hi @sproit. I'll see what I can find regarding Audio Manager Edge.

**Reply by mattias-kindborg-at-work** [reply, marked as answer]
The API is not intended for external usage. Perhaps you can use the [VAPIX Media clip API](https://www.axis.com/vapix-library/subjects/t10100065/section/t10036045/display), or perhaps you can use SIP paging or VAPIX paging ([transmit.cgi](https://www.axis.com/vapix-library/subjects/t10100065/section/t10036015/display?section=t10036015-t10003161))?

**Reply by ifordesmith** [reply]
Thanks Mattias - I'll take a look at the transmit.cgi option - I know that the VAPIX Media Clip API doesn't support what I'm trying to achieve.  Appreciate your input many thanks
